controllers/space/space_controller.go

```diff
@@ -712,6 +728,18 @@ func (r *Reconciler) setStatusNSTemplateSetUpdateFailed(logger logr.Logger, spac
 	return cause
 }
 
+// setStatusProvisionedNamespaces updates the provisioned namespace list status of the space if something changed.
+func (r *Reconciler) setStatusProvisionedNamespaces(space *toolchainv1alpha1.Space, provisionedNamespaces []toolchainv1alpha1.SpaceNamespace) (time.Duration, error) {
+	if reflect.DeepEqual(space.Status.ProvisionedNamespaces, provisionedNamespaces) {
+		// nothing changed
+		return 0, nil
+	}
+
+	// update provisioned namespace list with the one from the NSTemplateSet
+	space.Status.ProvisionedNamespaces = provisionedNamespaces
+	return requeueDelay, r.Client.Status().Update(context.TODO(), space)
```

Comment:
Why do you need to requeue if the status was updated?

Reply:
I did some more testing , and actually seems to not be needed. 
I've removed it in [6b62b93](https://github.com/codeready-toolchain/host-operator/pull/746/commits/6b62b931c5cda11b56dcad86ee562b9fcc3dbe00)

I added it initially because at some point the next call to updating the `Space.Status.Conditions to provisioned, and I was observing many errors of this type: 
`Operation cannot be fulfilled on ....  the object has been modified; please apply your changes to the latest version and try again` , but probably was related to some old version of my code 🤷‍♂️ 

Thanks!

Reply:
"Operation cannot be fulfilled on .... the object has been modified" can happen any moment. In this case we (should) return an error and got another reconciled requeued for us. No need to explicitly requeue in case of an error.